flickit-assessment-kit/src/main/java/org/flickit/assessment/kit/adapter/out/persistence/kitversion/KitVersionMapper.java

```diff
@@ -14,11 +14,13 @@
 @NoArgsConstructor(access = AccessLevel.PRIVATE)
 public class KitVersionMapper {
 
-    public static KitVersionJpaEntity createParamToJpaEntity(AssessmentKitJpaEntity kit, CreateKitVersionPort.Param param) {
+    public static KitVersionJpaEntity createParamToJpaEntity(Long id, AssessmentKitJpaEntity kit, CreateKitVersionPort.Param param) {
+        var statusVersion = param.status() == KitVersionStatus.ARCHIVE ? -id : param.status().getId();
```

Comment:
**Suggestion:** Refactor the logic for determining statusVersion to ensure it correctly handles all possible values of KitVersionStatus, avoiding misuse of the ID field. [general, importance: 6]
```suggestion
        var statusVersion = calculateStatusVersion(param.status(), id);
```